### ISSUE TITLE
d2i_ECPKParameters_fp throws lots of warnings

### ISSUE THREAD

**Issue by bbbrumley** [openssl/openssl#12443]
I think some of [these macros](https://github.com/openssl/openssl/blob/ce3080e931d77fda3bb4f2d923fcc6cec967d1a3/include/openssl/ec.h#L836) are stale. Below is master but I get the same warnings in 1.1.1, although the code seems to runs fine.

MWE:
```C
#include <openssl/ec.h>
#include <assert.h>

int main(int argc, char **argv) {
    EC_GROUP *group = NULL;
    FILE *fp = NULL;

    assert((fp = fopen("params.der", "rb")) != NULL);
    assert((group = d2i_ECPKParameters_fp(fp, NULL)) != NULL);
    fclose(fp);
    EC_GROUP_free(group);

    return 0;
}
```

```
bbrumley@arizona:/tmp$ rm foo; cc -o foo foo.c -lcrypto -L$HOME/.local/ssl/lib -I$HOME/.local/ssl/include
In file included from foo.c:2:0:
foo.c: In function ‘main’:
/home/bbrumley/.local/ssl/include/openssl/ec.h:841:35: warning: passing argument 2 of ‘ASN1_d2i_fp’ from incompatible pointer type [-Wincompatible-pointer-types]
     (EC_GROUP *)ASN1_d2i_fp(NULL, (char *(*)())d2i_ECPKParameters, (fp), \
                                   ^
foo.c:9:21: note: in expansion of macro ‘d2i_ECPKParameters_fp’
     assert((group = d2i_ECPKParameters_fp(fp, NULL)) != NULL);
                     ^~~~~~~~~~~~~~~~~~~~~
In file included from /home/bbrumley/.local/ssl/include/openssl/ec.h:23:0,
                 from foo.c:1:
/home/bbrumley/.local/ssl/include/openssl/asn1.h:689:7: note: expected ‘void * (*)(void **, const unsigned char **, long int)’ but argument is of type ‘char * (*)()’
 void *ASN1_d2i_fp(void *(*xnew) (void), d2i_of_void *d2i, FILE *in, void **x);
       ^~~~~~~~~~~
In file included from foo.c:2:0:
/home/bbrumley/.local/ssl/include/openssl/ec.h:842:29: warning: passing argument 4 of ‘ASN1_d2i_fp’ from incompatible pointer type [-Wincompatible-pointer-types]
                             (unsigned char **)(x))
                             ^
foo.c:9:21: note: in expansion of macro ‘d2i_ECPKParameters_fp’
     assert((group = d2i_ECPKParameters_fp(fp, NULL)) != NULL);
                     ^~~~~~~~~~~~~~~~~~~~~
In file included from /home/bbrumley/.local/ssl/include/openssl/ec.h:23:0,
                 from foo.c:1:
/home/bbrumley/.local/ssl/include/openssl/asn1.h:689:7: note: expected ‘void **’ but argument is of type ‘unsigned char **’
 void *ASN1_d2i_fp(void *(*xnew) (void), d2i_of_void *d2i, FILE *in, void **x);
       ^~~~~~~~~~~
In file included from foo.c:2:0:
/home/bbrumley/.local/ssl/include/openssl/ec.h:841:35: warning: passing argument 2 of ‘ASN1_d2i_fp’ from incompatible pointer type [-Wincompatible-pointer-types]
     (EC_GROUP *)ASN1_d2i_fp(NULL, (char *(*)())d2i_ECPKParameters, (fp), \
                                   ^
foo.c:9:21: note: in expansion of macro ‘d2i_ECPKParameters_fp’
     assert((group = d2i_ECPKParameters_fp(fp, NULL)) != NULL);
                     ^~~~~~~~~~~~~~~~~~~~~
In file included from /home/bbrumley/.local/ssl/include/openssl/ec.h:23:0,
                 from foo.c:1:
/home/bbrumley/.local/ssl/include/openssl/asn1.h:689:7: note: expected ‘void * (*)(void **, const unsigned char **, long int)’ but argument is of type ‘char * (*)()’
 void *ASN1_d2i_fp(void *(*xnew) (void), d2i_of_void *d2i, FILE *in, void **x);
       ^~~~~~~~~~~
In file included from foo.c:2:0:
/home/bbrumley/.local/ssl/include/openssl/ec.h:842:29: warning: passing argument 4 of ‘ASN1_d2i_fp’ from incompatible pointer type [-Wincompatible-pointer-types]
                             (unsigned char **)(x))
                             ^
foo.c:9:21: note: in expansion of macro ‘d2i_ECPKParameters_fp’
     assert((group = d2i_ECPKParameters_fp(fp, NULL)) != NULL);
                     ^~~~~~~~~~~~~~~~~~~~~
In file included from /home/bbrumley/.local/ssl/include/openssl/ec.h:23:0,
                 from foo.c:1:
/home/bbrumley/.local/ssl/include/openssl/asn1.h:689:7: note: expected ‘void **’ but argument is of type ‘unsigned char **’
 void *ASN1_d2i_fp(void *(*xnew) (void), d2i_of_void *d2i, FILE *in, void **x);
       ^~~~~~~~~~~
```

**Comment by romen** [comment]
@slontis could this have happened when some of the ASN1 functions were changed for constification? I think some of those were also backported to 1.1.1 and we don't have regression tests that use those functions probably!

**Comment by slontis** [comment]
Probably. Which is a good reason not try fixing this type of thing later. I cant recall who added the const.

**Comment by romen** [comment]
Reopening for `1.1.1`

**Comment by romen** [comment]
Fixed in `1.1.1` via 45487db.
